### DISCUSSION TITLE
Application switcher

### DISCUSSION THREAD

**Discussion by adocampo** [OpenVoiceOS/ovos-buildroot#92] [Ideas]
Yet many or most skills presents some information in the screen for a while and then shows the home screen back, there are anothers (music, video or news player) which can be running during a long period time and perhaps in that time user can ask for another skill (an alarm, the weather, etc), if the user ask something while one player is running, it ends on the home screen with music in the background and no other way to interact than stopping it via voice.

I think it could be of use to have a way to switch back to the media player via touch screen or even via voice. Perhaps a swipe up, showing a bottom bar with icons of the opened and running in background apps, _a la_ taskbar would be fair intuitive.

**Comment by JarbasAl** [comment]
there will be an icon in the taskbar, that currently only contains the camera app and also isn't pushed to latest image i think

but this is a bug, the previously active skill should be displayed not the homescreen, in this case the music player. I think of it like a stack, you add pages on top and remove pages from the top, when there are no more pages -> show homescreen

**Comment by AIIX** [comment]
https://www.youtube.com/watch?v=0Mp8sTfONN0&ab_channel=OpenVoiceOS we do have a concept of a voice application (voice skill with a dedicated homepage) launcher tho there is very little scope for it currently having a full-blown multitasking system, what is presented when an app is opened is totally up to the skill author at this point

**Comment by AIIX** [comment]
there are also ongoing discussions around skill interaction which might affect the project at some stage so best to also keep a lookout and participate at https://community.mycroft.ai/t/mycroft-skills-etiquette-how-should-they-interact/11401/18
